src/test/java/com/powsybl/caseserver/datasource/util/CaseDataSourceControllerTest.java

```diff
@@ -0,0 +1,173 @@
+import com.powsybl.commons.datasource.DataSource;
+import com.powsybl.iidm.import_.Importers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+import java.io.*;
+import java.net.URISyntaxException;
+
+import java.nio.file.*;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.asyncDispatch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.request;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
+@RunWith(SpringRunner.class)
+@EnableWebMvc
+@WebMvcTest(CaseDataSourceController.class)
+@ContextConfiguration(classes = {CaseDataSourceController.class, CaseDataSourceService.class})
+@TestPropertySource(properties = {"case-store-directory=test"})
+public class CaseDataSourceControllerTest {
+
+    private FileSystem fileSystem = Jimfs.newFileSystem();
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private CaseService caseService;
+
+    @Value("${case-store-directory:#{systemProperties['user.home'].concat(\"/cases\")}}")
+    private String rootDirectory;
+
+    private String cgmesName = "CGMES_v2415_MicroGridTestConfiguration_BC_BE_v2.zip";
+    private String fileName = "CGMES_v2.4.15_MicroGridTestConfiguration_BC_BE_v2/MicroGridTestConfiguration_BC_BE_DL_V2.xml";
+
+    private DataSource dataSource;
+
+    @Before
+    public void setUp() throws URISyntaxException, IOException {
+        Path path = fileSystem.getPath(rootDirectory);
+        if (!Files.exists(path)) {
+            try {
+                Files.createDirectories(path);
+            } catch (IOException e) {
+                fail();
+            }
+        }
+        caseService.setFileSystem(fileSystem);
+        //insert a cgmes in the FS
+        try (InputStream cgmesURL = getClass().getResourceAsStream("/" + cgmesName)) {
+            Path cgmes = path.resolve(cgmesName);
+            Files.copy(cgmesURL, cgmes, StandardCopyOption.REPLACE_EXISTING);
+        }
+        dataSource = Importers.createDataSource(Paths.get(getClass().getResource("/" + cgmesName).toURI()));
```

Comment:
Use ResourceDataSource instead. It has been designed to easy class resource loading as data source.